src/main/java/org/zalando/problem/spring/web/advice/validation/ConstraintViolationProblem.java

```diff
@@ -38,16 +40,16 @@
     public static final URI CONSTRAINT_VIOLATION = URI.create(CONSTRAINT_VIOLATION_VALUE);
 
     private final Optional<String> detail;
-    private final ImmutableList<Violation> violations;
+    private final List<Violation> violations;
 
-    public ConstraintViolationProblem(final ImmutableList<Violation> violations) {
-        this(Optional.empty(), violations);
+    public ConstraintViolationProblem(final List<Violation> violations) {
+        this(Optional.empty(), new ArrayList<>(violations));
     }
 
     @JsonCreator
-    private ConstraintViolationProblem(final Optional<String> detail, final ImmutableList<Violation> violations) {
```

Comment:
What's problem with copying and btw `ArrayList(Collection<? extends E> c)` is making a copy of the list

```
public ArrayList(Collection<? extends E> c) {
    elementData = c.toArray();
    if ((size = elementData.length) != 0) {
        // c.toArray might (incorrectly) not return Object[] (see 6260652)
        if (elementData.getClass() != Object[].class)
            elementData = Arrays.copyOf(elementData, size, Object[].class);
    } else {
        // replace with empty array.
        this.elementData = EMPTY_ELEMENTDATA;
    }
}
```

If you need *immutable* state you must copy